development/tools/pkg/diskscollector/collector_test.go

```diff
@@ -0,0 +1,198 @@
+package diskscollector
+
+import (
+	"errors"
+	"fmt"
+	"regexp"
+	"testing"
+	"time"
+
+	"github.com/kyma-project/test-infra/development/tools/pkg/diskscollector/automock"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	compute "google.golang.org/api/compute/v1"
+)
+
+const regexPattern = "^gke-gkeint.*[-]pvc[-]"
+const sampleDiskName = "gke-gkeint-abc-pvc-xyz"
+
+var (
+	diskNameRegex, _         = regexp.Compile(regexPattern)
+	filterFunc               = NewDiskFilter(diskNameRegex, 1) //age is 1 hour
+	timeNow                  = time.Now()
+	timeNowFormatted         = timeNow.Format(time.RFC3339Nano)
+	timeTwoHoursAgo          = timeNow.Add(time.Duration(-1) * time.Hour)
+	timeTwoHoursAgoFormatted = timeTwoHoursAgo.Format(time.RFC3339Nano)
+	nonEmptyUsers            = []string{"someUser"}
+)
+
+func TestNewDiskFilter(t *testing.T) {
+	//given
+	var testCases = []struct {
+		name                  string
+		expectedFilterValue   bool
+		diskName              string
+		diskCreationTimestamp string
+		diskUsers             []string
+	}{
+		{"Should filter matching disk", true, sampleDiskName, timeTwoHoursAgoFormatted, []string{}},
```

Comment:
Could you name parameters here? It makes test case a lot more visible.

Reply:
It's a bit verbose to me, but If it makes the code more readable, no problem.